cmd/lxd-to-incus/db.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"fmt"
+	"github.com/lxc/incus/shared"
+	"github.com/pierrec/lz4/v4"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// Uncompress the raft snapshot files in the given database directory.
+//
+// A backup will be created and kept around in case of errors.
+func migrateDatabase(dir string) error {
+	global := filepath.Join(dir, "global")
+
+	err := shared.DirCopy(global, global+".bak")
+	if err != nil {
+		return fmt.Errorf("Failed to backup database directory %q: %w", global, err)
+	}
+
+	files, err := ioutil.ReadDir(global)
+	if err != nil {
+		return fmt.Errorf("Failed to list database directory %q: %w", global, err)
+	}
+
+	for _, file := range files {
+		var timestamp uint64
+		var first uint64
+		var last uint64
+
+		if !file.Mode().IsRegular() {
+			continue
+		}
+
+		n, err := fmt.Sscanf(file.Name(), "snapshot-%d-%d-%d\n", &timestamp, &first, &last)
+		if err != nil || n != 3 {
+			continue
+		}
+
+		filename := filepath.Join(global, file.Name())
+		err = uncompress(filename)
+		if err != nil {
+			return fmt.Errorf("Failed to uncompress snapshot %q: %w", filename, err)
+		}
+	}
+
+	return nil
+}
+
+// Uncompress the given file, preserving its mode and ownership.
+func uncompress(zfilename string) error {
```

Comment:
Maybe `lz4Uncompress` so it's clear that it does just lz4?

Reply:
Fixed.